lib/ex_integrate.ex

```diff
@@ -4,11 +4,12 @@ defmodule ExIntegrate do
   """
 
   alias ExIntegrate.Boundary.PipelineRunner
```

Comment:
@garrettmichaelgeorge  why "boundary"? Just curious

Reply:
@samrose we had discussed adopting the conventions from *[Designing Elixir Systems with OTP](https://pragprog.com/titles/jgotp/designing-elixir-systems-with-otp/)*, where they namespace their functional core modules under `MyApp.Core.*` and their boundary modules under `MyApp.Boundary.* Since the PipelineRunner ends up performing IO operations by running steps, I treated it as a boundary module (a Pipeline itself, by contrast, is under `ExIntegrate.Core.Pipeline`.

But I have been a bit ambivalent about that namespacing idea lately. Outside that book, I haven't seen a single Elixir application that uses this convention. And while "core" and "boundary" are useful terms for considering the layers of the application, I don't know that they necessarily need to be enshrined in the code.

On the other hand, the conceptual separation between core and boundary modules has helped keep the code clean so far.

I had even considered using `ExIntegrate.Core.*` and `ExIntegrate.Boundary.*` as scaffolding until we got the main app structure the way we wanted, then removing the namespaces and simply using `ExIntegrate.Pipeline` and `ExIntegrate.PipelineRunner`.

If we were to follow the original idea through, we would probably want to move the files into `lib/ex_integrate/core/` and `lib/ex_integrate/boundary/` to match up with the module names.

Happy to discuss it more…I can't quite make up my mind about it.

Reply:
@garrettmichaelgeorge no that is great, I just spaced out on the origin of it so thanks